### PR TITLE
WIP: feat: Cosmos DB Generic Repository

### PR DESCRIPTION
**Pull request by DenzoNL**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

This PR adds a generic repository class that wraps the Cosmos DB Container, which provides simple CRUD functionality. A decorator is provided to inject the repository directly into a service provider, so one can simply do the following: 
```typescript
@InjectCosmosRepository(Cats) private readonly repository: AzureCosmosRepository<Cats>
```

I am submitting this as a WIP because this code could use some feedback and approval. It also needs tests and docs.

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by manekinekko** [comment]
Hi @DenzoNL you may wanna rebase your PR on master. I see some of the commits that are already in master.

**Comment by DenzoNL** [comment]
@manekinekko Done, although it does keep displaying those commits, it doesn't see them as changes. Not sure if I've done this correctly

**Comment by harveyramer** [comment]
I'm unclear what this functionality proposes that is not already available in Cosmos DB using the MongoDB api. I've been able to connect without drama using the instructions here. https://docs.microsoft.com/en-us/azure/cosmos-db/mongodb-mongoose

**Comment by manekinekko** [comment]
@harveyramer I guess the purpose of this PR is to abstract away to kind of driver that is being used by Cosmos DB. Of course, when using the mongodb driver with Cosmos DB, a developer can choose to use `@nestjs/mongoose mongoose` if they want.

Maybe @DenzoNL can give more feedback.

File: lib/cosmos-db/cosmos-db.providers.ts [hunk 1]
```diff
@@ -61,3 +63,19 @@ export function createAzureCosmosDbProviders(
   }));
   return providers;
 }
+
+export function getAzureCosmosRepositoryProvider(model: { dto: any }): Provider {
+  const provide = getAzureCosmosRepositoryToken(model.dto.name);
+  const o = {
```
Comment (manekinekko) [inline lib/cosmos-db/cosmos-db.providers.ts]: `o` should have an explicit name.

**Comment by manekinekko** [comment]
@DenzoNL would you mind updating the sample application and documentation with the changes from this PR?

**Comment by DenzoNL** [comment]
> Maybe @DenzoNL can give more feedback.

It's exactly what you said, just a generic repository class to abstract away some inner workings of Cosmos DB to scaffold simple CRUD functionality as fast as possible.


> @DenzoNL would you mind updating the [sample application](https://github.com/nestjs/azure-database/tree/master/sample/cosmos-db) and documentation with the changes from this PR?

I'll take a look at this tomorrow :) I haven't looked at it since I just integrated it into my own NestJS project directly, but I'll clean this up and update the docs. Thanks for the feedback!

**Comment by DenzoNL** [comment]
Too caught up at work to spend any time on this, unfortunately.

**Pull request closed without merging**